core/storage/question/gae_models.py

```diff
@@ -285,13 +285,19 @@ def get_question_skill_links_based_on_mastery_equidistributed_by_skill(
                 given skill ids, and the order of questions for the same skill
                 follows the difference between skill difficulty and mastery.
         """
+        if not isinstance(degrees_of_mastery, dict):
```

Comment:
You should not be throwing such an exception (Note this exception is already being thrown). What I meant is you need to handle this. One of the ways would be to set degrees_of_mastery to be an empty dict if its None, or the other way is to check if degrees_of_mastery exists before checking skill_id in degrees of mastery
i.e. 
In line 298 do:
```
if not (degrees_of_mastery or skill_id in degrees_of_mastery):
  degree_of_mastery = 0
else:
  degree_of_mastery = degrees_of_mastery[skill_id]
```

Reply:
This is handled in function get_user_skill_mastery, so I don't think it needs to be handled again.

Reply:
Done.